pkg/windows/calico.go

```diff
@@ -52,147 +137,235 @@ const (
 )
 
 // Setup creates the basic configuration required by the CNI.
-func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonconfig.Node, restConfig *rest.Config) (*CNIConfig, error) {
-	calicoKubeConfig := CalicoKubeConfig{
-		Server:               "https://127.0.0.1:6443",
-		CertificateAuthority: filepath.Join("c:\\", dataDir, "agent", "server-ca.crt"),
-		Token:                "",
-		Path:                 filepath.Join("c:\\", dataDir, "agent", CalicoKubeConfigName),
+func (c *Calico) Setup(ctx context.Context, nodeConfig *daemonconfig.Node, restConfig *rest.Config, dataDir string) error {
+	c.DataDir = dataDir
+
+	if err := c.initializeConfig(ctx, nodeConfig, restConfig); err != nil {
+		return err
+	}
+
+	if err := c.overrideCalicoConfigByHelm(restConfig); err != nil {
+		return err
+	}
+
+	if err := c.writeConfigFiles(nodeConfig.AgentConfig.CNIConfDir, nodeConfig.AgentConfig.NodeName); err != nil {
+		return err
 	}
 
-	client, err := coreClient(restConfig)
+	logrus.Info("Generating HNS networks, please wait")
+	return c.generateCalicoNetworks()
+}
+
+// initializeConfig sets the default configuration in CNIConfig
+func (c *Calico) initializeConfig(ctx context.Context, nodeConfig *daemonconfig.Node, restConfig *rest.Config) error {
+	c.CNICfg = &CalicoConfig{}
```

Comment:
It would appear as though this line isn't necessary since you're doing that and more on line 107,

Reply:
correct, good catch!